bridge-proxy/src/bridge-proxy.rs

```diff
@@ -86,49 +91,92 @@ pub trait BridgeProxyContract:
             tx_call
         };
 
+        let block_epoch = self.blockchain().get_block_epoch();
+        self.ongoing_execution(tx_id).set(block_epoch);
         tx_call.register_promise();
     }
 
+    #[endpoint(cancel)]
+    fn cancel(&self, tx_id: usize) {
+        let tx_start_epoch = self.ongoing_execution(tx_id).get();
+        let current_block_epoch = self.blockchain().get_block_epoch();
+        require!(
+            current_block_epoch - tx_start_epoch > DELAY_BEFORE_OWNER_CAN_CANCEL_TRANSACTION,
```

Comment:
Just to make this clear: we will be able to cancel transactions only after 10 epochs, right? Just to be able to clean the contract and return the funds, if, for example, no one wanted to execute that operation.

Reply:
I think 300 blocks would be more appropriate here.
Yes, if the refund fails for any reason (there are a few), the user would be able to claim the funds manually.
Will update the cooldown time.